calcom/cal.com
# feat(webhooks): pass webhook secret to `testTrigger`

## What does this PR do?



This PR addresses an issue with the webhook functionality in Cal.com where the "Secret" input is not being passed when using the "Webhook test" feature. 

I've noticed that when testing the webhook, the "Secret" field that was recently introduced is not being included in the test payload. This can make it difficult for developers to fully test the webhook's behavior, especially if they rely on the secret for authentication or other logic on their end.

https://www.loom.com/share/368ed245f76c44538b641dbd12947a09?sid=f0714f7e-c18c-455d-8afa-76351b541ef1

## Type of change




## How should this be tested?



- Are there environment variables that should be set? No.
- What are the minimal test data to have? None.
- What is expected (happy path) to have (input and output)? Happy path is to be able to receive the encoded secret in the test payload header. ("x-cal-signature-256").
- Any other important info that could help to test that PR. None.

Test guide:
1. Navigate to the Webhooks page in Cal.com's dashboard.
2. Add a webhook URL and specify a secret.
3. Click the "Test webhook" button.
4. Verify that the secret is now included in the test payload sent to the webhook URL (In order to verify the signature [read this](https://cal.com/docs/core-features/webhooks#verifying-the-authenticity-of-the-received-payload))

## Mandatory Tasks


## Checklist



- I haven't commented my code, particularly in hard-to-understand areas
- I haven't added tests that prove my fix is effective or that my feature works
